libs/libcommon/src/libcommon/orchestrator.py

```diff
@@ -306,7 +304,7 @@ def __post_init__(self) -> None:
                         name_field="split",
                     )  # Note that we use the cached content even the revision is different (ie. maybe obsolete)
                 for split_name in split_names:
-                    self.update(next_processing_step, config, split_name)
+                    self.update(next_processing_step, config, split_name, len(split_names))
```

Comment:
I think we should set the penalization to a value that reflects the total complexity of the dataset, not just the config's complexity.

For example, set it to the total number of splits of the dataset when we have it. Otherwise, if a dataset has 1000 configs with 2 splits each, the split-level jobs will have a penalization value of 2. It would be better to set it to 2000 (or the value available at that time, since we will progressively know the total number of splits the dataset contain).

Reply:
> Otherwise, if a dataset has 1000 configs with 2 splits each, the split-level jobs will have a penalization value of 2.

Actually, it will be the sum of job penalization +  parent penalization
https://github.com/huggingface/datasets-server/pull/2141/files#diff-45fde89a57a9a9a207231137c66a1ddf75b5830ba0eb7f504f87d87f921afb27R368

In this case, for a dataset with 1000 configs each with 2 splits, parents will have a penalization of 1000, and children will be 1002. Maybe it is enough since it will be used in the sort condition, also I thought I would add a default penalization by step to increase by 1 instead of 0 so that, we always have a descending value (In a small dataset with 1 config and 1 split, no job will have penalization).

Reply:
OK, I should have read the code more carefully. It seems a better approach, by the way, because if a dataset has two configs, one with two splits, and the other one with 1000 splits, the first config should not be penalized by the second one.

Reply:
> add a default penalization by step to increase by 1 instead of 0

I didn't get it, could you elaborate more on what it would change?

Reply:
I was suggesting it to have a progressive penalization (parents having more than children). But nevermind, I think it is better not to, because otherwise, it will always affect querying the predecessors before the children's processing steps (which is not what we want).

Reply:
OK? Let's start with the number of configs and splits, it should already help a lot